### PR TITLE
(rancher.dev) dhcp on ens3 interface and tests

### PR DESCRIPTION
**Pull request by cbarria**
basic network config and test for the rancher.dev cluster

### CODE REVIEW

File: hieradata/cluster/rancher/osfamily/RedHat/major/8.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+profile::nm::connections:
```
Comment (jhoblitt) [inline hieradata/cluster/rancher/osfamily/RedHat/major/8.yaml]: I'm not sure if this should be limited to EL8 as the network config should be the same for EL9.

Comment (cbarria) [reply hieradata/cluster/rancher/osfamily/RedHat/major/8.yaml]: where or how in the hierarchy can i put this to grab Alma 8 and 9, but not Centos 7? (i could copy and create a 9.yaml, but the idea is not to repeat the code right?)

Comment (jhoblitt) [reply hieradata/cluster/rancher/osfamily/RedHat/major/8.yaml]: There is no place in hierarchy which would apply this to EL7 as `profile::nm` is only included if the `os.release.major` is NOT `7`.

Comment (cbarria) [reply hieradata/cluster/rancher/osfamily/RedHat/major/8.yaml]: great! so i can go higher in the hierarchy and not affect EL7, thanks!

**Review by jhoblitt — APPROVED** [review]
LGTM.

**Pull request merged**